backend/src/main/java/ai/giskard/security/PermissionEvaluator.java

```diff
@@ -52,8 +52,10 @@ public boolean isGuestWithAnyRole(@NotNull Long projectId, @NotBlank String... r
     }
 
     public boolean canWriteProjectKey(@NotNull String projectKey) {
-        Project project = this.projectRepository.findOneByKey(projectKey).orElseThrow(() -> new EntityNotFoundException(Entity.PROJECT, projectKey));
-        return (isCurrentUser(project.getOwner().getLogin()) || SecurityUtils.isCurrentUserAdmin());
+        Project project = this.projectRepository.findOneByKey(projectKey)
+            .orElseThrow(() -> new EntityNotFoundException(Entity.PROJECT, projectKey));
+
+        return (isCurrentUserGuest(project) && this.canWrite()) || isCurrentUser(project.getOwner().getLogin()) || SecurityUtils.isCurrentUserAdmin();
```

Comment:
We also have a method called `canWriteProject` which doesn't replicate the code in `canWriteProjectKey`.

Ideally we should have a method `canWriteProjectKey(Project project)` that is called by both `canWriteProject(Long id)` and `canWriteProjectKey`.